gasmeter/gas_meter.go

```diff
@@ -0,0 +1,143 @@
+package gasmeter
+
+import (
+	"bufio"
+	"bytes"
+	"container/list"
+	"fmt"
+	"os"
+	"path/filepath"
+	"regexp"
+	"strconv"
+	"strings"
+	"testing"
+
+	"github.com/filecoin-project/chain-validation/box"
+	"github.com/filecoin-project/chain-validation/chain/types"
+)
+
+const ValidationDataEnvVar = "CHAIN_VALIDATION_DATA"
+
+type trackerElement struct {
+	receipt types.MessageReceipt
+}
+
+func (te *trackerElement) fileKey() string {
+	return fmt.Sprintf("%d", te.receipt.GasUsed.Int64())
+}
+
+type GasMeter struct {
+	tracker *list.List
+	T       testing.TB
+
+	// index in gasUnits of expected gas
+	gasIdx int
+	// slice of gas units used by the test
+	expectedGasUnits []int64
+}
+
+func NewGasMeter(t testing.TB) *GasMeter {
+	return &GasMeter{
+		tracker:          list.New(),
+		T:                t,
+		gasIdx:           0,
+		expectedGasUnits: LoadGasForTest(t),
+	}
+}
+
+func (gm *GasMeter) Track(receipt types.MessageReceipt) {
+	gm.tracker.PushBack(&trackerElement{
+		receipt: receipt,
+	})
+}
+
+func (gm *GasMeter) ExpectedGasUnit() (int64, bool) {
```

Comment:
Make the iterator-like functionality more clear with a name like `NextExpectedGas`